addons/stock_account/models/stock_valuation_layer.py

```diff
@@ -77,7 +77,26 @@ def _validate_analytic_accounting_entries(self):
     def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
         if 'unit_cost' in fields:
             fields.remove('unit_cost')
-        return super().read_group(domain, fields, groupby, offset, limit, orderby, lazy)
+        # A product's current value may not align with the value of its SVLs tied to stock moves:
+        # E.g., product cost manually changed after reception -> adjusting SVL has no stock_move_id
+        # and thus the value returned by the read_group will not reflect up-to-date valuation.
+        does_query_move_location_dest = False
+        if any('value' in field for field in fields):
```

Comment:
seems too large to just check if the string contains value anywhere, shouldn't this be at least `field.startwith('value')` and at best something like `field.partition(':')[0] == 'value'`?